packages/typegpu/src/data/attributes.ts

```diff
@@ -92,6 +99,12 @@ export type IsBuiltin<T> = ExtractAttributes<T>[number] extends []
     ? true
     : false;
 
+export type HasCustomInterpolate<T> = ExtractAttributes<T>[number] extends []
+  ? false
+  : ExtractAttributes<T>[number] extends Interpolate<InterpolationType>
+    ? true
+    : false;
+
```

Comment:
I don't think we need this type utility as of this moment. `HasCustomLocation` was useful for determining whether to add a location attribute or not on the type level when used by pipelines. In this case, interpolation is always explicit, so we don't need this utility.

```suggestion
```

Reply:
will remove